orangewidget/widget.py

```diff
@@ -962,6 +965,31 @@ def __updateSavedGeometry(self):
             # Note: This should always be stored as bytes and not QByteArray.
             self.savedWidgetGeometry = bytes(self.saveGeometry())
 
+    def sizeHint(self):
+        sh = QSize()
+        # boxes look nice on macOS with starting width/height 4
+        width = 4
+        height = 4
+
+        if self.want_message_bar:
+            msh = self.statusBar().sizeHint()
+            height += msh.height()
+        if self.want_control_area:
+            csh = self.controlArea.sizeHint()
+            width += csh.width()
+            height += csh.height()
+            if self.buttons_area_orientation:
+                bsh = self.buttonsArea.sizeHint()
+                height += bsh.height()
```

Comment:
This sizeHint computation is quite complex. Why wouldn't superclass sizeHint() work if you then:
- for width, subtract mainArea's sizeHint width?
- for height, do max(height, 500)?